operators/derive/to_es.py

```diff
@@ -0,0 +1,68 @@
+import dataflows as DF
+from dataflows.helpers.resource_matcher import ResourceMatcher
+from dataflows_elasticsearch import dump_to_es
+
+from conf import settings
+from srm_tools.logger import logger
+
+
+def set_es_item_types(resources=None):
+    def func(package):
+        matcher = ResourceMatcher(resources, package.pkg)
+        for resource in package.pkg.descriptor['resources']:
+            if matcher.match(resource['name']):
```

Comment:
Eventually we'll have to set these explicitly on fields and not generally.
The reason is that different fields need to get indexed differently in ES - 
- slugs will probably get indexed as keywords (i.e. treat them as a single string, no text analysis done)
- names will get indexed as text (i.e. so that full text search can be done on them)
etc.

Reply:
@akariv yes. I see table schema-elasticsearch supports custom mappings https://github.com/frictionlessdata/tableschema-elasticsearch-py/blob/65c18467fe23f1508030668d09588daf8b2badbb/tableschema_elasticsearch/mappers.py#L13 via a custom `mapping_generator_cls` and I also see the various existing hooks to control how the ES document is created https://github.com/frictionlessdata/tableschema-elasticsearch-py#mappings but I don't see a way to say something as simple as "map our fieldname ID of type string to an ES keyword". Am I missing something?

Reply:
No, you're not missing anything - the selection of indexing analyzer and other options is done outside of the driver, as it's very application specific.
A usage example: https://github.com/hasadna/migdar-data-pipelines/blob/master/datapackage_pipelines_migdar/flows/dump_to_es.py#L10

Reply:
I'm replacing this with a MappingGenerator